### ISSUE TITLE
keepalived.data and keepalived.stats datas are not synced error

### ISSUE THREAD

**Issue by dmachard** [mehdy/keepalived-exporter#115]
with old version (Keepalived v1.3.9 on Ubuntu 2018), the following error occured in random way and the metric keepalived_up is set to 0 but in fact the keepalived is running properly:

`keepalived_exporter[80560]: time="2023-04-14T15:52:37+02:00" level=error msg="keepalived.data and keepalived.stats datas are not synced"`

It's possible to just warn this error to support old version too ?

collector.go

```
if len(vrrpData) != len(vrrpStats) {
	logrus.Warn("keepalived.data and keepalived.stats datas are not synced: ", len(vrrpData), len(vrrpStats))
	// return nil, errors.New("keepalived.data and keepalived.stats datas are not synced")
}

if len(vrrpData) == len(vrrpStats) {
...
}
```

**Comment by clwluvw** [comment]
This probably is because stats or data file is created with more delay than the other so the wait time should be increased. We might make the initial wait configurable.
cc: @mehdy

**Comment by mehdy** [comment]
I think so. It'd be good to consider this and back off for a while to let them sync. And it's a good idea as well to log these things to be more traceable.

**Comment by dmachard** [comment]
You refer to these hardcoded time values in utils/files.go ?

```
b := backoff.NewExponentialBackOff()
b.InitialInterval = 10 * time.Millisecond
b.MaxElapsedTime = 2 * time.Second
```

**Comment by clwluvw** [comment]
Yes and probably we should add the parsing phase in the backoff loop so it will retry until the `data` and `stats` are synced.

**Comment by dmachard** [comment]
I will make a pull request proposal, 
but I need to make more investigations on this issue before.

**Comment by yinzuobiao** [comment]
Hello, I used keepalived-exporter when using prometheus to monitor the keepavlied service. I found this error in the log. My keepavlied version is v2.2.4, keepalived-exporter version is 1.3.0, and prometheus version is 2.37.6 . I'm new and don't know the code, but until now the problem is still unsolved, the service is still reporting an error, I hope it can get everyone's help, and finally solve this problem:
ul 05 16:25:39 fscc-lb-haproxy-01 keepalived-exporter[265011]: time="2023-07-05T16:25:39+08:00" level=error msg="No data found to be exported " error="keepalived.data and keepalived.stats datas are not synced"
Jul 05 16:35:18 fscc-lb-haproxy-01 keepalived-exporter[265011]: time="2023-07-05T16:35:18+08:00" level=error msg="keepalived.data and keepalived. stats data are not synced"
Jul 05 16:35:18 fscc-lb-haproxy-01 keepalived-exporter[265011]: time="2023-07-05T16:35:18+08:00" level=error msg="No data found to be exported " error="keepalived.data and keepalived.stats datas are not synced"
Jul 05 16:38:45 fscc-lb-haproxy-01 keepalived-exporter[265011]: time="2023-07-05T16:38:45+08:00" level=error msg="keepalived.data and keepalived. stats data are not synced"
Jul 05 16:38:45 fscc-lb-haproxy-01 keepalived-exporter[265011]: time="2023-07-05T16:38:45+08:00" level=error msg="No data found to be exported " error="keepalived.data and keepalived.stats datas are not synced"
